pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -152,7 +159,10 @@ func (m *mergeQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]
 // queriers. It also adds the `idLabelName` and if present in the original
 // results the original `idLabelName`.
 func (m *mergeQuerier) LabelNames() ([]string, storage.Warnings, error) {
-	labelNames, warnings, err := m.mergeDistinctStringSlice(func(ctx context.Context, q storage.Querier) ([]string, storage.Warnings, error) {
+	log, ctx := spanlogger.New(m.ctx, "mergeQuerier.LabelNames")
+	defer log.Span.Finish()
+	labelNames, warnings, err := m.mergeDistinctStringSlice(ctx, func(ctx context.Context,
```

Comment:
[nit] Please rollback this change.

Reply:
you are right, this context is not used in callback function so it's not necessary to pass it to this method. done